core/src/Kokkos_Complex.hpp

```diff
@@ -242,75 +242,108 @@ class complex {
     re_ = v;
   }
 
+  template<typename InputRealType>
   KOKKOS_INLINE_FUNCTION
-  complex<RealType>& operator += (const complex<RealType>& src) {
+  typename std::enable_if<std::is_convertible<InputRealType,RealType>::value,
```

Comment:
Given that the ``+=`` (and others members) are not overloaded then the ``std::enable_if`` is not necessary and introduces extra compiler work.
Instead just return the local ``complex`` and add to the function body
``static_assert( std::is_convertible<InputRealType,RealType>::value , "Input must be convertible" );``

Reply:
Done. Plz, See the next commit.